pkg/snapshot/v1alpha1/snapshot.go

```diff
@@ -211,6 +211,13 @@ func (s *snapshot) Delete() (*v1alpha1.CASSnapshot, error) {
 		return nil, err
 	}
 
+	// Return success for delete snapshot request in case volume snapshot belongs to a
+	// migrated CSI based persistent volume. Later as part of migration snapshot
+	// can be migrated to CSI snapshots using a migration or manual steps
+	if string(pv.Spec.CSI.Driver) == string(v1alpha1.CSIDriverName) {
```

Comment:
This will give nil pointer error for external provisioned volumes with ISCSI spec.

Reply:
ahh missed to add that check, added now